animation_nodes/nodes/boolean/switch_node.py

```diff
@@ -27,5 +35,24 @@ def assignType(self, dataType):
             self.refresh()
 
     def getExecutionCode(self, required):
-        if "output" in required: yield "output = ifTrue if condition else ifFalse"
-        if "other" in required:  yield "other = ifFalse if condition else ifTrue"
+        if self.useList:
+            if "dataList" in required: yield "dataList = self.execute_list(conditions, ifTrue, ifFalse)"
+        else:
+            if "output" in required: yield "output = ifTrue if condition else ifFalse"
+            if "other" in required:  yield "other = ifFalse if condition else ifTrue"
+
+    def execute_list(self, conditions, ifTrue, ifFalse):
+        outputs = []
+        for condition in conditions:
+            if condition:
+                outputs.append(ifTrue)
+            else:
+                outputs.append(ifFalse)
+        if not isList(self.assignedType):
+            listDataType = toListDataType(self.assignedType)
+            return self.outputs[0].correctValue(outputs)[0]
+        else:
+            outs = []
+            for i in range(len(conditions)):
+                outs.extend(outputs[i])
+            return self.outputs[0].correctValue(outs)[0]
```

Comment:
A virtual list can be used here. Are you familiar with those?

Reply:
Sorry, i am not familiar with virtual list, and have never used it.

Reply:
Virtual Lists are described in https://docs.animation-nodes.com/release_notes/v2_1/#virtual-lists. Also see #1363. A lot of nodes use them, for instance, look at any of the Geometry nodes. Let me know if you need further help in understanding them.

Reply:
Thank you very much! i will try my best to understand them.